packages/firestore/src/core/sync_engine.ts

```diff
@@ -671,22 +746,68 @@ export class SyncEngine implements RemoteSyncer, SharedClientStateSyncer {
   }
 
   // PORTING NOTE: Multi-tab only
-  applyTargetState(
+  async applyTargetState(
     targetId: TargetId,
     state: QueryTargetState,
     error?: FirestoreError
   ): Promise<void> {
-    // TODO(multitab): Implement this
-    return Promise.resolve();
+    // Apply the target state if the target is either actively being listened to
+    // or a limbo target change (which could then apply to any target).
+    if (
+      this.queryViewsByTarget[targetId] ||
+      this.limboTargetIdGenerator.covers(targetId)
+    ) {
+      switch (state) {
+        case 'current':
+        case 'not-current': {
+          const changes = await this.localStore.getNewDocumentChanges();
+          return this.emitNewSnapsAndNotifyLocalStore(
+            changes,
+            state === 'current'
```

Comment:
Er... this doesn't seem right since it's applying the per-target state to all document changes (which could apply to multiple targets).